app/src/main/java/com/zeapo/pwdstore/utils/PasswordRepository.kt

```diff
@@ -31,6 +31,18 @@ open class PasswordRepository protected constructor() {
             p1.name.compareTo(p2.name, ignoreCase = true)
         }),
 
+        RECENTLY_USED(Comparator { p1: PasswordItem, p2: PasswordItem ->
+            val recentHistory = Application.instance.getSharedPreferences("recent_password_history", android.content.Context.MODE_PRIVATE)
+            val timeP1 = recentHistory.getString(p1.file.absolutePath)
+            val timeP2 = recentHistory.getString(p2.file.absolutePath)
+            when {
+                timeP1 != null && timeP2 != null -> timeP2.compareTo(timeP1)
```

Comment:
This could be simplified using `compareBy` and `nullsLast`, but I could also add that later.

Reply:
I'll leave it to you, I have no idea how to use those...